### ISSUE TITLE
No code highlighting inside QUOTE macros

### ISSUE THREAD

**Issue by dedmen** [acemod/language-arma-atom#82]
I am using QUOTE(code) for the onLoad scripts in ui configs. But its not highlighting any code words.
It should look like this:
https://github.com/CBATeam/CBA_A3/blob/master/addons/diagnostic/gui.hpp#L14 (Github highlighting does it correctly)
but it looks like this:
![Image of Atom code](http://image.prntscr.com/image/8af32a6dd1974ee7a7addb287f3129d7.png)

As you can see encasing the code in {} fixes it a little. But doesn't fix the QUOTE macro itself not highlighting.

**Comment by thojkooi** [comment]
This is mostly due to the fact we do not highlight sqf keywords in config files. The quote bit seems weird, maybe we can adjust our regex.

**Comment by jonpas** [comment]
`*.cpp` and `*.hpp` files are using C++ highlighting. By design, but if we can modify C++ highlighting to fit that wouldn't be bad.

**Comment by thojkooi** [comment]
Do we not add the CBA macros to those files?

**Comment by jonpas** [comment]
No, everything is for SQF only. @dedmen Set your Atom to "Arma Config" when you open one of those files and see if it changes anything.

**Comment by dedmen** [comment]
Above screenshots were made with Arma config grammar. Setting it to SQF highlights everything correctly.
Though its kinda unusual to have a bunch of script in a config file. But also highlighting of safezone\* commands, which are often used in UI's, don't highlight in configs which is a little more important.

**Comment by jonpas** [comment]
Works like that in C++ header highlighter which we use, and we don't do any SQF highlighting at all in configs. Would just mean copying them to the `cfg.json` as well.
